### ISSUE TITLE
[Bug] Debian 11 - GPG error: https://apt.hestiacp.com bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5

### ISSUE THREAD

**Issue by asimzeeshan** [hestiacp/hestiacp#3716]
### Describe the bug

Spin up a brand new Debian 11 VM
Install the control panel using the script provided [here](https://hestiacp.com/install.html)

Getting the following error
```
                _   _           _   _        ____ ____
               | | | | ___  ___| |_(_) __ _ / ___|  _ \
               | |_| |/ _ \/ __| __| |/ _` | |   | |_) |
               |  _  |  __/\__ \ |_| | (_| | |___|  __/
               |_| |_|\___||___/\__|_|\__,_|\____|_|

                          Hestia Control Panel
                                  1.7.7
                            www.hestiacp.com

========================================================================

Thank you for downloading Hestia Control Panel! In a few moments,
we will begin installing the following components on your server:

   - NGINX Web / Proxy Server
   - Apache Web Server (as backend)
   - Multi-PHP Environment
   - Bind DNS Server
   - Exim Mail Server + ClamAV + SpamAssassin
   - Dovecot POP3/IMAP Server+ Sieve
   - MariaDB Database Server
   - Vsftpd FTP Server
   - Firewall (iptables) + Fail2Ban Access Monitor

========================================================================


Would you like to continue with the installation? [Y/N]: Y
Installation backup directory: /root/hst_install_backups/260620230218
Installation log file: /root/hst_install_backups/hst_install-260620230218.log

Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait... W: GPG error: https://apt.hestiacp.com bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5
E: The repository 'https://apt.hestiacp.com bullseye InRelease' is not signed.
-
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait...
E: Unable to locate package hestia
E: Unable to locate package hestia-nginx
E: Unable to locate package hestia-php


========================================================================

[ * ] Configuring system settings...
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/httpd-prerotate/*': No such file or directory
sed: can't read /etc/systemd/timesyncd.conf: No such file or directory
Failed to enable unit: Unit file systemd-timesyncd.service does not exist.
Failed to start systemd-timesyncd.service: Unit systemd-timesyncd.service not found.
[ * ] Configuring Hestia Control Panel...
cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
rm: cannot remove '/var/log/hestia': Is a directory
ln: failed to create symbolic link '/usr/local/hestia/log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/system.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/nginx-error.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/auth.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/backup.log': No such file or directory
chmod: cannot access '/usr/local/hestia/log': No such file or directory
chmod: cannot access '/var/log/hestia/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/packages': No such file or directory
sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/dns/': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/unassigned/index.html': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/skel/document_errors/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/firewall': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/api': No such file or directory
[ * ] Generating default self-signed SSL certificate...
hst-install-debian.sh: line 1264: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
[ * ] Adding SSL certificate to Hestia Control Panel...
sed: -e expression #1, char 3: unexpected `,'
sed: -e expression #1, char 1: unknown command: `,'
cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
[ * ] Enable SFTP jail...
Error: can't enable sftp jail
```

If I use this method to pull the key, it doesn't work

```
root@hestiacp:~# apt-key adv --keyserver keyserver.ubuntu.com --recv-keys A189E93654F0B0E5
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Executing: /tmp/apt-key-gpghome.zOkaGoToat/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys A189E93654F0B0E5
gpg: key A189E93654F0B0E5: "Hestia Control Panel <info@hestiacp.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```

If I use the newer method, it doesn't work as well

```
root@hestiacp:~# gpg --no-default-keyring --keyserver keyring.debian.org --recv-keys A189E93654F0B0E5
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
```

**It says no valid OpenPGP data found, that's weird**

Upon running `apt update` manually, I get the following error

```
root@hestiacp:~# apt update
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://deb.debian.org/debian bullseye InRelease
Hit:3 http://deb.debian.org/debian bullseye-updates InRelease
Hit:4 http://deb.debian.org/debian bullseye-backports InRelease
Hit:5 https://nginx.org/packages/mainline/debian bullseye InRelease
Hit:6 https://packages.sury.org/apache2 bullseye InRelease                                             t
Hit:7 https://packages.sury.org/php bullseye InRelease
Get:8 https://apt.hestiacp.com bullseye InRelease [59.3 kB]
Get:9 https://dlm.mariadb.com/repo/mariadb-server/10.11/repo/debian bullseye InRelease [3882 B]
Err:8 https://apt.hestiacp.com bullseye InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5
Reading package lists... Done
W: GPG error: https://apt.hestiacp.com bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5
E: The repository 'https://apt.hestiacp.com bullseye InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

### Tell us how to replicate the bug

1. Spin new Debian 11 VM
2. Login to SSH
3. Download the script wget https://raw.githubusercontent.com/hestiacp/hestiacp/release/install/hst-install.sh
4. Run it with parameters as advised on [this page](https://hestiacp.com/install.html)
5. Get the error
6. Delete the directory "/usr/local/hestia" as [advised here](https://forum.hestiacp.com/t/error-while-installing-hestia/6881) (otherwise the installer cannot be run again)
7. Try to find a solution to the NO_PUBKEY issue but failing everytime

### Which components are affected by this bug?

Control Panel Installation or Upgrade

### Hestia Control Panel Version

1.7.7

### Operating system

Debian 11

### Log capture

```shell
Already shared everything above
```

**Comment by jaapmarcus** [comment]
Try:

```
gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys A189E93654F0B0E5
```

```
gpg: keybox '/usr/share/keyrings/hestia-keyring.gpg' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key A189E93654F0B0E5: public key "Hestia Control Panel <info@hestiacp.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
```

**Comment by jaapmarcus** [comment]
Just made a test install:

Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait...|
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait... 
Extracting templates from packages: 100%


========================================================================

[ * ] Configuring system settings...
[ * ] Configuring Hestia Control Panel...
[ * ] Generating default self-signed SSL certificate...
[ * ] Adding SSL certificate to Hestia Control Panel...
[ * ] Enable SFTP jail...
[ * ] Create admin account...
[ * ] Configuring NGINX...
[ * ] Updating Cloudflare IP Ranges for Nginx...
[ * ] Configuring Apache Web Server...
[ * ] Install PHP 5.6...
[ * ] Install PHP 7.0...

And it seems to be working fine

**Comment by jaapmarcus** [comment]
Maybe your ip has been blocked send an email to info@hestiacp.com with your public server ip

**Comment by asimzeeshan** [comment]
Just tried adding the key again and it **FAILED** :(

```
root@hestiacp:~# gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys A189E93654F0B0E5
gpg: WARNING: unacceptable HTTP redirect from server was cleaned up
gpg: keyserver receive failed: Server indicated a failure
```

**Comment by asimzeeshan** [comment]
Email sent to info@hestiacp.com with my IP information

**Comment by jaapmarcus** [comment]
Could not find any matches in the firewall 

How ever added your ip to the whitelist

**Comment by asimzeeshan** [comment]
Trying on Ubuntu 22.04 LTS

```
root@ hestiacp:~# cat /etc/*-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

Same issue as Debian 11, pasting the logs here

```

                _   _           _   _        ____ ____
               | | | | ___  ___| |_(_) __ _ / ___|  _ \
               | |_| |/ _ \/ __| __| |/ _` | |   | |_) |
               |  _  |  __/\__ \ |_| | (_| | |___|  __/
               |_| |_|\___||___/\__|_|\__,_|\____|_|

                          Hestia Control Panel
                                  1.7.7
                            www.hestiacp.com

========================================================================

Thank you for downloading Hestia Control Panel! In a few moments,
we will begin installing the following components on your server:

   - NGINX Web / Proxy Server
   - Apache Web Server (as backend)
   - Multi-PHP Environment
   - Bind DNS Server
   - Exim Mail Server + ClamAV + SpamAssassin
   - Dovecot POP3/IMAP Server+ Sieve
   - MariaDB Database Server
   - Vsftpd FTP Server
   - Firewall (iptables) + Fail2Ban Access Monitor

========================================================================


Would you like to continue with the installation? [Y/N]: Y
Installation backup directory: /root/hst_install_backups/260620231125
Installation log file: /root/hst_install_backups/hst_install-260620231125.log

Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait... W: GPG error: https://apt.hestiacp.com jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5
E: The repository 'https://apt.hestiacp.com jammy InRelease' is not signed.
\
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait...
E: Unable to locate package hestia
E: Unable to locate package hestia-nginx
E: Unable to locate package hestia-php


========================================================================

[ * ] Configuring system settings...
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/httpd-prerotate/*': No such file or directory
[ * ] Configuring Hestia Control Panel...
cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
ln: failed to create symbolic link '/usr/local/hestia/log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/system.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/nginx-error.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/auth.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/backup.log': No such file or directory
chmod: cannot access '/usr/local/hestia/log': No such file or directory
chmod: cannot access '/var/log/hestia/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/packages': No such file or directory
sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/dns/': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/unassigned/index.html': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/templates/web/skel/document_errors/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/firewall': No such file or directory
cp: cannot stat '/usr/local/hestia/install/common/api': No such file or directory
[ * ] Generating default self-signed SSL certificate...
hst-install-ubuntu.sh: line 1295: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
[ * ] Adding SSL certificate to Hestia Control Panel...
sed: -e expression #1, char 3: unexpected `,'
sed: -e expression #1, char 1: unknown command: `,'
cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
[ * ] Enable SFTP jail...
Error: can't enable sftp jail
```


And the key doesn't get added here as well, same issue as Debian 11

```
root@hestiacp:~# gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys A189E93654F0B0E5
gpg: WARNING: unacceptable HTTP redirect from server was cleaned up
gpg: keyserver receive failed: Server indicated a failure
```

**Comment by asimzeeshan** [comment]
It is worth mentioning that I have tried it **on the same box** (the IPv4/IPv6 of which I have already shared via Ticket#911800) and you have already mentioned that the IP is whitelisted (not blocked)

**Comment by jaapmarcus** [comment]
What doe you see if run: 

wget https://apt.hestiacp.com/

It should just download an index.html with the contains "apt.hestiacp.com" and not Cloudflare blocked message?

**Comment by asimzeeshan** [comment]
```
root@hestiacp:~# curl https://apt.hestiacp.com/
<h1>apt.hestiacp.com</h1>
<script defer src="https://static.cloudflareinsights.com/beacon.min.js/v52afc6f149f6479b8c77fa569edb01181681764108816" integrity="sha512-jGCTpDpBAYDGNYR5ztKt4BQPGef1P0giN6ZGVUi835kFF88FOmmn8jBQWNgrNd8g/Yu421NdgWhwQoaOPFflDw==" data-cf-beacon='{"rayId":"7dd53c527897039a","version":"2023.4.0","b":1,"token":"e6f6f252670947fa93c1485ae0fbca6a","si":100}' crossorigin="anonymous"></script>
```

**Comment by jaapmarcus** [comment]
Cloudflare blocks the traffic...

**Comment by asimzeeshan** [comment]
@jaapmarcus How can I resolve this because for me the logical first step is resolving the NO_PUBKEY issue and later figuring out anything else

**Comment by jaapmarcus** [comment]
The pub key is working fine how ever apt.hestiacp.com doesn't work ... Because Cloudflare blocks the traffic to it...

**Comment by jaapmarcus** [comment]
Ip was still blocked by Cloudflare firewall.. Created a new rule for see it now works.. 

<img width="578" alt="Screenshot 2023-06-26 at 14 18 19" src="https://github.com/hestiacp/hestiacp/assets/9754650/5dae83f0-e756-4e17-8416-d9345bc6de83">

Please try again

**Comment by asimzeeshan** [comment]
Finally, it worked

for the record, I have manually added the key using this 
```
root@hestiacp:~# gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver keyserver.ubuntu.com --recv-keys A189E93654F0B0E5
```

instead of this

```
root@hestiacp:~#  gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys A189E93654F0B0E5
```

**Comment by harlanmukdis** [comment]
> Trying on Ubuntu 22.04 LTS
> 
> ```
> root@ hestiacp:~# cat /etc/*-release
> DISTRIB_ID=Ubuntu
> DISTRIB_RELEASE=22.04
> DISTRIB_CODENAME=jammy
> DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS"
> PRETTY_NAME="Ubuntu 22.04.2 LTS"
> NAME="Ubuntu"
> VERSION_ID="22.04"
> VERSION="22.04.2 LTS (Jammy Jellyfish)"
> VERSION_CODENAME=jammy
> ID=ubuntu
> ID_LIKE=debian
> HOME_URL="https://www.ubuntu.com/"
> SUPPORT_URL="https://help.ubuntu.com/"
> BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
> PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
> UBUNTU_CODENAME=jammy
> ```
> 
> Same issue as Debian 11, pasting the logs here
> 
> ```
> 
>                 _   _           _   _        ____ ____
>                | | | | ___  ___| |_(_) __ _ / ___|  _ \
>                | |_| |/ _ \/ __| __| |/ _` | |   | |_) |
>                |  _  |  __/\__ \ |_| | (_| | |___|  __/
>                |_| |_|\___||___/\__|_|\__,_|\____|_|
> 
>                           Hestia Control Panel
>                                   1.7.7
>                             www.hestiacp.com
> 
> ========================================================================
> 
> Thank you for downloading Hestia Control Panel! In a few moments,
> we will begin installing the following components on your server:
> 
>    - NGINX Web / Proxy Server
>    - Apache Web Server (as backend)
>    - Multi-PHP Environment
>    - Bind DNS Server
>    - Exim Mail Server + ClamAV + SpamAssassin
>    - Dovecot POP3/IMAP Server+ Sieve
>    - MariaDB Database Server
>    - Vsftpd FTP Server
>    - Firewall (iptables) + Fail2Ban Access Monitor
> 
> ========================================================================
> 
> 
> Would you like to continue with the installation? [Y/N]: Y
> Installation backup directory: /root/hst_install_backups/260620231125
> Installation log file: /root/hst_install_backups/hst_install-260620231125.log
> 
> Adding required repositories to proceed with installation:
> 
> [ * ] NGINX
> [ * ] PHP
> [ * ] Apache2
> [ * ] MariaDB
> [ * ] Hestia Control Panel
> 
> Updating currently installed packages, please wait... W: GPG error: https://apt.hestiacp.com jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A189E93654F0B0E5
> E: The repository 'https://apt.hestiacp.com jammy InRelease' is not signed.
> \
> The installer is now downloading and installing all required packages.
> NOTE: This process may take 10 to 15 minutes to complete, please wait...
> E: Unable to locate package hestia
> E: Unable to locate package hestia-nginx
> E: Unable to locate package hestia-php
> 
> 
> ========================================================================
> 
> [ * ] Configuring system settings...
> cp: cannot stat '/usr/local/hestia/install/deb/logrotate/httpd-prerotate/*': No such file or directory
> [ * ] Configuring Hestia Control Panel...
> cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
> chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
> ln: failed to create symbolic link '/usr/local/hestia/log': No such file or directory
> touch: cannot touch '/usr/local/hestia/log/system.log': No such file or directory
> touch: cannot touch '/usr/local/hestia/log/nginx-error.log': No such file or directory
> touch: cannot touch '/usr/local/hestia/log/auth.log': No such file or directory
> touch: cannot touch '/usr/local/hestia/log/backup.log': No such file or directory
> chmod: cannot access '/usr/local/hestia/log': No such file or directory
> chmod: cannot access '/var/log/hestia/*': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/packages': No such file or directory
> sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/templates/web/': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/templates/dns/': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/templates/web/unassigned/index.html': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/templates/web/skel/document_errors/*': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/firewall': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/common/api': No such file or directory
> [ * ] Generating default self-signed SSL certificate...
> hst-install-ubuntu.sh: line 1295: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
> [ * ] Adding SSL certificate to Hestia Control Panel...
> sed: -e expression #1, char 3: unexpected `,'
> sed: -e expression #1, char 1: unknown command: `,'
> cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
> [ * ] Enable SFTP jail...
> Error: can't enable sftp jail
> ```
> 
> And the key doesn't get added here as well, same issue as Debian 11
> 
> ```
> root@hestiacp:~# gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys A189E93654F0B0E5
> gpg: WARNING: unacceptable HTTP redirect from server was cleaned up
> gpg: keyserver receive failed: Server indicated a failure
> ```

**Comment by harlanmukdis** [comment]
yeah same this issue in ubuntu 20.04 LTS

**Comment by jaapmarcus** [comment]
Run the above checks if ip has been blocked by CF please send an email to the listed email...

**Comment by TresIqus94** [comment]
Writing a solution to this issue for those having issues

`gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver hkp://185.125.188.26 --recv-keys A189E93654F0B0E5`

or

`gpg --no-default-keyring --keyring /usr/share/keyrings/hestia-keyring.gpg --keyserver 185.125.188.26 --recv-keys A189E93654F0B0E5`

For whatever reason the domain doesn't seem to work because it resolves to both 185.125.188.26 and 185.125.188.27
